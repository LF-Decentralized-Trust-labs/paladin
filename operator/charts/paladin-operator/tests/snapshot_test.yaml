suite: Snapshot tests
tests:
  - it: Basic snapshot test
    when: snapshot
    asserts:
      - matchSnapshot: {}
    set:
      unittest: true
  - it: Change namespace
    when: snapshot
    asserts:
      - matchSnapshot: {}
    set:
      unittest: true
      operator.namespace: test
      
