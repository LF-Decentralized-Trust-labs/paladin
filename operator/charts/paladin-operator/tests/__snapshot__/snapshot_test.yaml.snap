Basic snapshot test:
  1: |
    raw: Welcome to Paladin!
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      name: paladin-operator
      namespace: paladin
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: Role
      name: paladin-operator
    subjects:
      - kind: ServiceAccount
        name: paladin-operator
        namespace: paladin
  3: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: Role
    metadata:
      name: paladin-operator
      namespace: paladin
    rules:
      - apiGroups:
          - core.paladin.io
        resources:
          - smartcontracts
          - besus
          - paladins
          - besugeneses
        verbs:
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - core.paladin.io
        resources:
          - smartcontracts/status
          - besus/status
          - paladins/status
          - besugeneses/status
        verbs:
          - get
      - apiGroups:
          - apps
        resources:
          - statefulsets
        verbs:
          - get
          - list
          - create
          - delete
          - update
          - patch
          - watch
      - apiGroups:
          - ""
        resources:
          - services
          - configmaps
          - secrets
          - persistentvolumeclaims
        verbs:
          - get
          - list
          - create
          - delete
          - update
          - patch
          - watch
      - apiGroups:
          - policy
        resources:
          - poddisruptionbudgets
        verbs:
          - get
          - list
          - create
          - delete
          - update
          - patch
          - watch
  4: |
    apiVersion: v1
    data:
      config.json: |
        {
          "paladin": {
            "image": "paladin:test",
            "labels": {
              "app": "paladin"
            }
          },
          "besu": {
            "image": "hyperledger/besu:latest",
            "labels": {
              "app": "besu"
            }
          },
          "postgres": {
            "image": "postgres:latest"
          }
        }
    kind: ConfigMap
    metadata:
      name: paladin-operator
      namespace: paladin
  5: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations: null
      labels:
        app.kubernetes.io/name: paladin-operator
        control-plane: controller-manager
      name: paladin-operator
      namespace: paladin
    spec:
      replicas: null
      selector:
        matchLabels:
          app.kubernetes.io/name: paladin-operator
      template:
        metadata:
          labels:
            app.kubernetes.io/name: paladin-operator
        spec:
          containers:
            - command:
                - /manager
              env:
                - name: WATCH_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: CONFIG_PATH
                  value: /etc/config
              image: paladin-operator:latest
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: 8081
                  scheme: HTTP
              name: paladin-operator
              ports:
                - containerPort: 80
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                  scheme: HTTP
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts:
                - mountPath: /etc/config/config.json
                  name: config
                  subPath: config.json
          securityContext:
            runAsNonRoot: true
          serviceAccountName: paladin-operator
          volumes:
            - configMap:
                name: paladin-operator
              name: config
  6: |
    apiVersion: v1
    kind: Service
    metadata:
      name: paladin-operator
      namespace: paladin
    spec:
      ports:
        - name: http
          port: 80
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/name: paladin-operator
      type: ClusterIP
  7: |
    apiVersion: v1
    automountServiceAccountToken: true
    kind: ServiceAccount
    metadata:
      name: paladin-operator
      namespace: paladin
Change namespace:
  1: |
    raw: Welcome to Paladin!
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      name: paladin-operator
      namespace: test
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: Role
      name: paladin-operator
    subjects:
      - kind: ServiceAccount
        name: paladin-operator
        namespace: test
  3: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: Role
    metadata:
      name: paladin-operator
      namespace: test
    rules:
      - apiGroups:
          - core.paladin.io
        resources:
          - smartcontracts
          - besus
          - paladins
          - besugeneses
        verbs:
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - core.paladin.io
        resources:
          - smartcontracts/status
          - besus/status
          - paladins/status
          - besugeneses/status
        verbs:
          - get
      - apiGroups:
          - apps
        resources:
          - statefulsets
        verbs:
          - get
          - list
          - create
          - delete
          - update
          - patch
          - watch
      - apiGroups:
          - ""
        resources:
          - services
          - configmaps
          - secrets
          - persistentvolumeclaims
        verbs:
          - get
          - list
          - create
          - delete
          - update
          - patch
          - watch
      - apiGroups:
          - policy
        resources:
          - poddisruptionbudgets
        verbs:
          - get
          - list
          - create
          - delete
          - update
          - patch
          - watch
  4: |
    apiVersion: v1
    data:
      config.json: |
        {
          "paladin": {
            "image": "paladin:test",
            "labels": {
              "app": "paladin"
            }
          },
          "besu": {
            "image": "hyperledger/besu:latest",
            "labels": {
              "app": "besu"
            }
          },
          "postgres": {
            "image": "postgres:latest"
          }
        }
    kind: ConfigMap
    metadata:
      name: paladin-operator
      namespace: test
  5: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations: null
      labels:
        app.kubernetes.io/name: paladin-operator
        control-plane: controller-manager
      name: paladin-operator
      namespace: test
    spec:
      replicas: null
      selector:
        matchLabels:
          app.kubernetes.io/name: paladin-operator
      template:
        metadata:
          labels:
            app.kubernetes.io/name: paladin-operator
        spec:
          containers:
            - command:
                - /manager
              env:
                - name: WATCH_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: CONFIG_PATH
                  value: /etc/config
              image: paladin-operator:latest
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: 8081
                  scheme: HTTP
              name: paladin-operator
              ports:
                - containerPort: 80
                  name: http
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                  scheme: HTTP
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts:
                - mountPath: /etc/config/config.json
                  name: config
                  subPath: config.json
          securityContext:
            runAsNonRoot: true
          serviceAccountName: paladin-operator
          volumes:
            - configMap:
                name: paladin-operator
              name: config
  6: |
    apiVersion: v1
    kind: Service
    metadata:
      name: paladin-operator
      namespace: test
    spec:
      ports:
        - name: http
          port: 80
          protocol: TCP
          targetPort: http
      selector:
        app.kubernetes.io/name: paladin-operator
      type: ClusterIP
  7: |
    apiVersion: v1
    automountServiceAccountToken: true
    kind: ServiceAccount
    metadata:
      name: paladin-operator
      namespace: test
